pkg/minio/knowledgebase.go

```diff
@@ -77,14 +87,19 @@ func (m *Minio) SaveTextChunks(ctx context.Context, kbUID string, chunks map[Chu
 			func() {
 				func(chunkUID ChunkUIDType, chunkContent ChunkContentType) {
 					defer wg.Done()
-					filePathName := m.GetChunkPathInKnowledgeBase(kbUID, string(chunkUID))
 
-					err := m.UploadBase64File(ctx, config.Config.Minio.BucketName, filePathName, base64.StdEncoding.EncodeToString(chunkContent), "text/plain")
+					path := filepath.Join(chunkBasePath(kbUID, fileUID), string(chunkUID)) + ".txt"
+					err := m.UploadBase64File(ctx, config.Config.Minio.BucketName, path, base64.StdEncoding.EncodeToString(chunkContent), "text/plain")
 					if err != nil {
 						logger.Error("Failed to upload chunk after retries", zap.String("chunkUID", string(chunkUID)), zap.Error(err))
 						errorUIDChan <- ChunkError{ChunkUID: string(chunkUID), ErrorMessage: err.Error()}
 						return
 					}
+
+					// Add successful chunk to destinations map
+					mu.Lock()
+					destinations[string(chunkUID)] = path
```

Comment:
🗒️ In parallel to the destination being returned in the converted file creation method, we return here a map of chunk UID -> destination, removing the exported `GetChunkPathInKnowledgeBase` and simplifying the code in the worker.